main.tf

```
@@ -389,6 +393,10 @@ resource "aws_subnet" "private" {
   ipv6_cidr_block = var.enable_ipv6 && length(var.private_subnet_ipv6_prefixes) > 0 ? cidrsubnet(aws_vpc.this[0].ipv6_cidr_block, 8, var.private_subnet_ipv6_prefixes[count.index]) : null
 
   tags = merge(
+    length(var.private_subnet_names) > 0 ?
+    {
+      "Name" = element(var.private_subnet_names, count.index)
```

In your approach would make sense to keep the az value appended? e.g.

```suggestion
      "Name" = format(
        "${element(var.private_subnet_names, count.index)}-%s",
        element(var.azs, count.index),
```

---

@v-rosa Thank you for the suggestion. I tested this out and it worked well. I rebased my branch onto latest master, and while I did so I also re-wrote my original commit to have this suggestion.

---

I'm sorry, I don't get it. Why would you append AZ when you want to control subnet names manually? Is this the whole point of this PR - to be able to set custom names to subnets?

---

If you define your network with multiple AZs, e.g. 3, you would get per custom name, 3 subnets with the same custom name. Also the naming wouldn't be consistent with the other subnets created using default naming.

---

@v-rosa I think `default naming` and `custom naming` are two separate things, you usually want to pick just one approach. If you want to include AZ to your `custom naming` you just do something like this
```hcl
module "vpc" {
  source  = "terraform-aws-modules/vpc/aws"
  version = "3.14.4"

  name = "development"
...
  public_subnet_names = ["mysuper-custom-name-us-west-2a", "another-name-us-west-2b"]
}
``` 
In my case I don't want to include AZ to my custom subnet schema, I should be able to do that since it's custom :)